fix(pagination): add pageSizeInputDisabled prop to Pagination

tw15egan opened the pull request:
Closes https://github.com/carbon-design-system/carbon/issues/6055

This PR now ties all `Pagination` inputs / buttons to the `disabled` prop, rather than having 3 separate props ton control them individually. This also deprecates the `pageInputDisabled` prop.

#### Changelog

**Changed**

- `disabled` prop now controls both `select` elements, as well as the `back` and `forward` buttons.
- `pageInputDisabled` prop now will throw depraction warnings 

#### Testing / Reviewing

Go to `Pagination`, enable the `disabled` prop, and make sure all inputs are disabled

tw15egan committed:
fix(pagination): add pageSizeInputDisabled prop to Pagination

aagonzales reviewed (approved):
Looks good to me 👍

tw15egan committed:
chore(tests): update snapshots

emyarod commented on packages/react/src/components/Pagination/Pagination-story.js:
```diff
@@ -23,6 +23,10 @@ const props = () => ({
   totalItems: number('Total number of items (totalItems)', 103),
   pagesUnknown: boolean('Total number of items unknown (pagesUnknown)', false),
+  pageSizeInputDisabled: boolean(
+    'Disable page size input (pageInputDisabled)',
```
```suggestion
    'Disable page size input (pageSizeInputDisabled)',
```

emyarod reviewed (commented):
the prop name in the knobs section is the same as the other input knob

joshblack commented:
Looks great after Andrew's suggestions above!

tw15egan committed:
fix(pagination): fix story prop name

tw15egan commented:
@emyarod good catch, updated 👍

emyarod reviewed (commented):
are there instances where each input is disabled separately from the others or should the logic be combined for certain component states?

tw15egan commented:
I'm not sure of the total use cases when the pagination is disabled, so I didn't want to combine any of the disabling in case they are used in different scenarios.

I think eventually we may want to go that route, but might be best to wait until v11 and have them all use the disabled attribute?

emyarod commented:
just looking for clarification on the component usage before adding the extra prop, since if it is supposed to just be linked with the regular `disabled` prop we would be adding in a new prop now and then deprecating it in v11 right?

joshblack commented:
bump @tw15egan if you have a sec to respond to the last comment 👀

tw15egan commented:
I don't have any insight into that, but if you think we should just tie the disabled states together I'm fine with that 👍

emyarod commented:
yeah, similar to https://github.com/carbon-design-system/carbon/pull/6112#issuecomment-632193921 I think if the disabled states are meant to be linked then it makes sense not to expand the API surface area

tw15egan committed:
fix(pagiantion): disable page size input with existing prop

tw15egan committed:
feat(pagination): tie all inputs to disabled prop

tw15egan committed:
Merge branch 'master' into page-size-disabled

tw15egan commented:
@emyarod @joshblack @aagonzales I updated this PR to tie all inputs to the `disabled` prop and added a deprecation warning to `pageInputDisabled` prop.

tw15egan committed:
Merge branch 'master' into page-size-disabled

aagonzales reviewed (approved):
Cool, yeah I'm not sure how people are using the disabled features with pagination. But one way to find out is to change this and see if anyone complains. I think it should be fine though I can't really image why some might be disabled but not others. Seems like products would use this disabled feature mostly for when there isn't any page to paginate to.

tw15egan commented:
@aagonzales agreed, I reached out to the original issue creator and they agreed with that assumption as well

<img width="757" alt="Screen Shot 2020-06-09 at 3 23 53 PM" src="https://user-images.githubusercontent.com/11928039/84190708-37492b80-aa4c-11ea-9878-abbc60ea5295.png">

https://github.com/carbon-design-system/carbon/issues/6055#issuecomment-635606624

tw15egan committed:
Merge branch 'master' into page-size-disabled

joshblack commented on packages/react/src/components/Pagination/Pagination.js:
```diff
@@ -125,9 +126,12 @@ export default class Pagination extends Component {
-    pageInputDisabled: PropTypes.bool,
+    pageInputDisabled: deprecate(
+      PropTypes.bool,
+      `\nThe prop \`pageInputDisabled\` for Pagination has been deprecated, as the feature of \`pageInputDisabled\` has been combined with the general \`disabled\` prop.`
```
```suggestion
      `The prop \`pageInputDisabled\` for Pagination has been deprecated, as the feature of \`pageInputDisabled\` has been combined with the general \`disabled\` prop.`
```

tw15egan committed:
Update packages/react/src/components/Pagination/Pagination.js

Co-authored-by: Josh Black <josh@josh.black>

joshblack committed:
Merge branch 'master' into page-size-disabled

tw15egan committed:
Merge branch 'master' into page-size-disabled

emyarod reviewed (approved):
looks good to me

tw15egan committed:
Merge branch 'master' into page-size-disabled

s100 commented:
Following this change, it seems like it's now impossible to use `<Pagination>` without raising a deprecation warning. If we set the `pageInputDisabled` prop to `true` or `false`, we get the warning. If we set it to `null` or `undefined`, or omit it entirely, it defaults to `false` and we get the warning. This is a problem for us because our unit tests fail if warnings or errors are raised. Any suggestions?
